src/features/world/scenes/BeachScene.ts

```diff
@@ -361,6 +326,45 @@ export class BeachScene extends BaseScene {
       }
     });
 
+    const piratePotionEquipped = [
```

Comment:
Consider placing this code within the `if (hasFeatureAccess(this.gameState, "PIRATE_CHEST")) {` branch.

This is useful in case a bug arises, then only beta testers are impacted 👍